docs/stdlib/generic.rst

```diff
@@ -44,10 +44,16 @@ Generic
 
 -----------
 
+.. note::
+
+    In EdgeQL, any value can be compared to another as long as they are
+    both the same type.
 
 .. eql:operator:: eq: anytype = anytype -> bool
 
-    Compare two values for equality.
+    Compares two values of :eql:type:`anytype` for equality.
```

Comment:
This could be read to mean that you can compare any two values regardless of type. I see the note at the top, but it would be nice to encapsulate that here, and that might even obviate the need for the note altogether.
```suggestion
    Compares two values of compatible types for equality.
```

I would suggest this phrasing be applied globally to the comparison operators.

Might be nice to follow the existing examples with a couple showing attempts to compare values of incompatible types (bool = integer or string = number). This could also be applied to the other comparison operators.